full/src/test/java/apoc/load/LoadHtmlTest.java

```diff
@@ -53,22 +56,64 @@ public void setup() {
     }
 
     @Test
-    public void testQueryAll(){
-        Map<String, Object> query = map("metadata", "meta", "h2", "h2");
+    public void testParseGeneratedJs() {
+        testCallGeneratedJsWithBrowser("FIREFOX");
+        testCallGeneratedJsWithBrowser("CHROME");
+    }
 
-        testResult(db, "CALL apoc.load.html($url,$query, $config)", map("url",new File("src/test/resources/wikipedia.html").toURI().toString(), "query", query, "config", Collections.emptyMap()),
+    @Test
+    public void testWithWaitUntilAndOneElementNotFound() {
+        testCall(db, "CALL apoc.load.html($url,$query,$config)",
+                map("url",new File("src/test/resources/html/wikipediaWithJs.html").toURI().toString(),
+                        "query", map("elementExistent", "strong", "elementNotExistent", ".asdfgh"),
+                        "config", map("browser", "CHROME", "wait", 5)),
                 result -> {
-                    Map<String, Object> row = result.next();
-                    Map<String, Object> value = (Map<String, Object>) row.get("value");
-
-                    List<Map<String, Object>> metadata = (List<Map<String, Object>>) value.get("metadata");
-                    List<Map<String, Object>> h2 = (List<Map<String, Object>>) value.get("h2");
-
-                    assertEquals(asList(RESULT_QUERY_METADATA).toString().trim(), metadata.toString().trim());
-                    assertEquals(asList(RESULT_QUERY_H2).toString().trim(), h2.toString().trim());
+                    Map<String, Object> value = (Map<String, Object>) result.get("value");
+                    List<Map<String, Object>> notExistent = (List<Map<String, Object>>) value.get("elementNotExistent");
+                    List<Map<String, Object>> existent = (List<Map<String, Object>>) value.get("elementExistent");
+                    assertTrue(notExistent.isEmpty());
+                    assertEquals(1, existent.size());
+                    final Map<String, Object> tag = existent.get(0);
+                    assertEquals("This is a new text node", tag.get("text"));
+                    assertEquals("strong", tag.get("tagName"));
                 });
     }
 
+    @Test
+    public void testWithBaseUriConfig() {
+        Map<String, Object> query = map("urlTest", ".urlTest");
+
+        final String baseUri = new File("src/test/resources").toURI().toString();
+        testCall(db, "CALL apoc.load.html($url,$query, $config)",
+                map("url", new File("src/test/resources/html/wikipediaWithJs.html").toURI().toString(),
+                        "query", query,
+                        "config", map("baseUri", baseUri)),
+                result -> {
+                    Map<String, Object> value = (Map<String, Object>) result.get("value");
+                    final List<Map<String, Object>> urlTestList = (List<Map<String, Object>>) value.get("urlTest");
+                    urlTestList.forEach(tag -> {
```

Comment:
maybe compare `urlTestList` to a predefined `List<Map<...>>`? It may be easier to read than the current switch block